src/main/java/seedu/address/logic/parser/SortCommandParser.java

```diff
@@ -20,31 +22,36 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public Command parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
-        }
-
         try {
             String[] keywords = args.split(" ", 3);
-            String travelPlanObjectType = keywords[1].substring(1);
-            String sortKeyword = keywords[2];
+            String travelPlanObjectType = keywords[OBJECT_TYPE_POSITION].substring(1);
 
             switch (travelPlanObjectType) {
             case Activity.TPO_WORD:
-                return new SortActivityCommand(sortKeyword);
+                if (keywords.length < 3) {
+                    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            SortActivityCommand.MESSAGE_USAGE));
+                }
```

Comment:
Should you check this outside of the switch case and just throw SortCommand.MESSAGE_USAGE instead?

Reply:
I make it so its like 
"sort -activity " will show the error message specific to activity
"sort -accommodation" will show error message specific to accommodation etc.
do you think its ok?